app/gui/view/execution-mode-selector/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+        shape.set_size(Vector2::new(10.0, 11.0));
+        shape.set_x(max_width / 2.0 - PLAY_BUTTON_OFFSET);
+    }
+
+    fn update_position(&self, camera: &Camera2d) {
+        let screen = camera.screen();
+        let x = -screen.width / 2.0 + OVERALL_WIDTH / 2.0;
+        let y = screen.height / 2.0 - HEIGHT / 2.0;
+        self.inner_root.set_x(x.round());
+        self.inner_root.set_y(y.round());
+        error!("update_position x: {}, y: {}", x, y);
+    }
+
+    fn set_entries(&self, entries: Rc<Vec<ExecutionMode>>) {
+        let provider = ensogl_list_view::entry::AnyModelProvider::from(entries.clone_ref());
+        self.dropdown.set_entries(provider);
+        self.dropdown.set_selected(0);
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+
+// =============================
+// === ExecutionModeDropdown ===
+// =============================
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "ExecutionModeDropdown"
+    }
+
+    fn new(app: &Application) -> Self {
+        let scene = &app.display.default_scene;
+
+        let display_object = display::object::Instance::new();
+        let inner_root = display::object::Instance::new();
+        let background = background::View::new();
+        let play_icon = play_icon::View::new();
+        let dropdown = ensogl_drop_down_menu::DropDownMenu::new(app);
+
+        display_object.add_child(&inner_root);
+        inner_root.add_child(&dropdown);
+        inner_root.add_child(&play_icon);
+        inner_root.add_child(&background);
+
+        scene.layers.panel.add(&inner_root);
+        scene.layers.panel.add(&dropdown);
+        scene.layers.panel.add(&play_icon);
+
+        dropdown.set_label_layer(&scene.layers.panel_text);
+
+        Self::init_dropdown(&dropdown);
+        Self::init_background(OVERALL_WIDTH, &background);
+        Self::init_play_icon(OVERALL_WIDTH, &play_icon);
+
+        Self { display_object, background, play_icon, dropdown, inner_root }
+    }
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(
+        network: &enso_frp::Network,
+        frp: &<Self as ensogl::application::frp::API>::Private,
+        app: &Application,
+        model: &Model,
+        _style_watch: &ensogl::display::shape::StyleWatchFrp,
+    ) {
+        let scene = &app.display.default_scene;
+        let camera = scene.camera();
+        let dropdown = &model.dropdown;
+        let play_icon = &model.play_icon;
+        let input = &frp.input;
+        let output = &frp.output;
+
+        frp::extend! { network
+            init <- source();
+
+            // Layout
+            let camera_changed = scene.frp.camera_changed.clone_ref();
+            update_position <- any(init, camera_changed);
+            eval_ update_position ([model, camera] {
+                model.update_position(&camera);
+            });
```

Comment:
you don't need curly braces here and this can be one-line expressions.

Reply:
Technically, there is a difference, because here we discard potential return value from `update_position`.

But more important is, that  rustfmt formats too long lambdas this way, and we are meant to apply same formatting rules to macros.

Reply:
Good point with return value!
Regarding rustfmt, without curly braces I believe rustfmt would format that one-line.